src/profile/external-identities/OAuth2Link.tsx

```diff
@@ -0,0 +1,154 @@
+import { Clickable } from '@terra-ui-packages/components';
+import React from 'react';
+import { useState } from 'react';
+import { LazyClipboardButton } from 'src/components/ClipboardButton';
+import { ButtonPrimary } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { EcmLinkAccountResponse } from 'src/libs/ajax/ExternalCredentials';
+import colors from 'src/libs/colors';
+import { withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import * as Utils from 'src/libs/utils';
+import { OAuth2Callback, OAuth2Provider } from 'src/profile/external-identities/OAuth2Providers';
+import { SpacedSpinner } from 'src/profile/SpacedSpinner';
+
+const styles = {
+  clickableLink: {
+    display: 'inline',
+    color: colors.accent(),
+    cursor: 'pointer',
+    fontWeight: 500,
+  },
+  idLink: {
+    container: {
+      display: 'grid',
+      marginBottom: '0.6rem',
+      border: `1px solid ${colors.dark(0.55)}`,
+      borderRadius: 4,
+    },
+    linkContentTop: (hasBottom) => ({
+      display: 'grid',
+      rowGap: '0.6rem',
+      backgroundColor: colors.light(0.2),
+      padding: '1.2rem',
+      borderRadius: hasBottom ? '4px 4px 0 0' : 4,
+    }),
+    linkContentBottom: {
+      padding: '1.2rem',
+    },
+    linkName: {
+      fontSize: 18,
+      fontWeight: 700,
+      marginBottom: '0.6rem',
+      display: 'inline',
+    },
+    linkDetailLabel: {
+      fontWeight: 700,
+      marginBottom: '0.6rem',
+      marginRight: '1.2rem',
+    },
+  },
+};
+
+interface OAuth2LinkProps {
+  queryParams: { state?: string; code?: string };
+  provider: OAuth2Provider;
+}
+export const OAuth2Link = (props: OAuth2LinkProps) => {
+  const {
+    queryParams: { state, code },
+    provider,
+  } = props;
+
+  const signal = useCancellation();
+  const [accountInfo, setAccountInfo] = useState<EcmLinkAccountResponse>();
+  const [accountLoaded, setAccountLoaded] = useState<boolean>(false);
+  const callbacks: Array<OAuth2Callback> = ['oauth_callback', 'ecm-callback']; // ecm-callback is deprecated, but still needs to be supported
+  const [isLinking] = useState(
+    callbacks.includes(Nav.getCurrentRoute().name) && state && JSON.parse(atob(state)).provider === provider.key
+  );
```

Comment:
If there's no setter, does this need to be stored in state?

```suggestion
  const isLinking = callbacks.includes(Nav.getCurrentRoute().name) && state && JSON.parse(atob(state)).provider === provider.key;
```